src/viz/interactivity/viewport-features/ViewportFeaturesGenerator.ts

```diff
@@ -58,43 +152,18 @@ export class ViewportFeaturesGenerator {
     return this.getMVTViewportFilteredFeatures(selectedTiles, properties);
   }
 
-  private async getGeoJSONLayerFeatures(properties: string[] = []) {
-    const features = this.getGeoJSONFeatures();
-    let viewport: Viewport;
-
-    try {
-      viewport = this.getViewport();
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.warn('Viewport not ready');
+  // #region MVTLayer
+  private getSelectedTiles(): ViewportTile[] {
+    if (!this._deckLayer || !this._deckLayer.state) {
       return [];
     }
 
-    const currentFrustumPlanes = viewport.getFrustumPlanes();
-
-    return features
-      .filter(feature => {
-        if (!feature.properties) {
-          return false;
-        }
-
-        return checkIfGeometryIsInsideFrustum(
-          transformGeometryCoordinatesToCommonSpaceByViewport(
-            feature.geometry as GeometryData,
-            viewport
-          ),
-          currentFrustumPlanes
-        );
-      })
-      .map(feature => {
-        return selectPropertiesFrom(feature.properties as Record<string, unknown>, properties);
-      })
-      .flat();
+    return this._deckLayer.state.tileset.selectedTiles;
   }
 
   private getMVTViewportFilteredFeatures(selectedTiles: ViewportTile[], properties: string[]) {
     const currentFrustumPlanes = this.getViewport().getFrustumPlanes();
-    const featureCache = new Set<number>();
+    const featureCache = new Set<string>(); // don't assume just number (the most common use case, eg cartodb_id)
```

Comment:
Could be better new Set<string|number>() ? what do you think?

Reply:
That's the idea, but with an explicit conversion from number (integer expected) always to string. Let me have a look at it, to see if I can refactor it